src/insights_client/__init__.py

```diff
@@ -351,37 +349,62 @@ def _main():
     attempt to collect and upload with new, then current, then rpm
     if an egg fails a phase never try it again
     """
+    set_up_logging()
+    logger.debug("Verifying initial eggs to load config.")
+
+    validated_eggs = []
+    for egg in [STABLE_EGG, RPM_EGG]:
+        logger.debug("GPG verifying initial egg %s..." % (egg,))
+        if gpg_validate(egg):
+            logger.debug("GPG verification successful. Loading initial egg: %s" % (egg,))
+            validated_eggs.append(egg)
+        else:
+            logger.debug("WARNING: GPG verification failed. Not loading initial egg: %s" % (egg,))
     # sort rpm and stable eggs after verification
-    validated_eggs = sorted_eggs(list(filter(gpg_validate, [STABLE_EGG, RPM_EGG])))
+    validated_eggs = sorted_eggs(validated_eggs)
     # if ENV_EGG was specified and it's valid, add that to front of sys.path
     #  so it can be loaded initially. keep it in its own var so we don't
     #  pass it to run_phase where we load it again
+    logger.debug("Verifying initial ENV_EGG: %s..." % (ENV_EGG,))
     if gpg_validate(ENV_EGG):
+        logger.debug("GPG verification successful. Loading initial egg: %s" % (ENV_EGG,))
         valid_env_egg = [ENV_EGG]
     else:
+        logger.debug("WARNING: GPG verification failed. Not loading initial egg: %s" % (ENV_EGG,))
         valid_env_egg = []
 
     if not validated_eggs and not valid_env_egg:
-        sys.exit("No GPG-verified eggs can be found")
+        sys.exit("No GPG-verified initial eggs can be found")
 
     # ENV egg comes first
     sys.path = valid_env_egg + validated_eggs + sys.path
 
     try:
+        sys_path = ":".join(sys.path)
+        logger.debug("Importing the insights module from %s..." % (sys_path,))
         # flake8 complains because these imports aren't at the top
         import insights
+        egg_path = os.path.dirname(insights.__path__[0])
+        logger.debug("Loaded initial egg: %s" % ( egg_path,))
+
         from insights.client import InsightsClient
         from insights.client.phase.v1 import get_phases
         from insights.client.config import InsightsConfig
 
         # Add the insights-config here
         try:
-            config = InsightsConfig(_print_errors=True).load_all()
+            logger.debug("Initializing InsightsConfig.")
```

Comment:
I don't think this is interesting log information.

Reply:
🤔  Well, the InsightsConfig initialization is pretty heavy. It provides an information that the modules are imported and we’re going to actually initialize the class that parses environment values, a config file and command line arguments, merges them etc. Again, it’s the last message you’ll see if the config initialization fails.